packages-next/admin-ui/src/system/generateAdminUI.ts

```diff
@@ -52,15 +52,24 @@ export const generateAdminUI = async (
 ) => {
   // Nuke any existing files in our target directory
   await fs.remove(projectAdminPath);
+  const publicDirectory = Path.join(projectAdminPath, 'public');
 
   if (config.images) {
-    const publicDirectory = Path.join(projectAdminPath, 'public');
     await fs.mkdir(publicDirectory, { recursive: true });
     const storagePath = Path.resolve(config.images.local?.storagePath ?? './public/images');
     await fs.mkdir(storagePath, { recursive: true });
     await fs.symlink(storagePath, Path.join(publicDirectory, 'images'), 'junction');
   }
 
+  if (config.files) {
+    if (!fs.existsSync(Path.join(projectAdminPath, 'public'))) {
```

Comment:
Checking if something exists in the file system and then immediately reading it or etc. is generally bad because it might might at the time of the exists call and not at a read or write or something (not a concern here really but more generally so it's best to avoid it) and mkdir will not error if it already exists so the exists check is just adding another file system operation, could you change this so there is one call to mkdir like 

```ts
if (config.images || config.files) {
      await fs.mkdir(publicDirectory, { recursive: true });
}
```

Reply:
ah good to know, will do thxx Mitchell 🎉